patches/api/0468-Added-EntityDamageDoorEvent-and-implemented-into-Ser.patch

```diff
@@ -0,0 +1,132 @@
+new file mode 100644
+index 0000000000000000000000000000000000000000..0e631ec59726a586260ac95ba793c4544c10d632
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/EntityDamageDoorEvent.java
+@@ -0,0 +1,120 @@
++package io.papermc.paper.event.entity;
++
++import com.google.common.base.Preconditions;
++import org.bukkit.block.data.BlockData;
++import org.bukkit.entity.Entity;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Range;
++
++/**
++ * Called for every tick an entity is damaging a door.
++ * <p>
++ * This event is only called when a new level of progress
++ * has been reached by the damaging entity.
++ */
++public class EntityDamageDoorEvent extends EntityEvent implements Cancellable {
++
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++    private final BlockData damagedDoor;
++    private boolean cancelled;
++    private int breakTime;
++    private final int doorBreakTime;
++    private boolean playEffect;
++
++    public EntityDamageDoorEvent(final @NotNull Entity what, final @NotNull BlockData damagedDoor, final int breakTime, final int doorBreakTime, boolean playEffect) {
++        super(what);
++        this.damagedDoor = damagedDoor;
++        this.breakTime = breakTime;
++        this.doorBreakTime = doorBreakTime;
++        this.playEffect = playEffect;
++    }
++
++    /**
++     * Get the {@link BlockData} representing the Door
++     * currently being damaged by the entity.
++     *
++     * @return The door being damaged.
++     */
++    @NotNull
++    public BlockData getDamagedDoor() {
++        return this.damagedDoor;
++    }
++
++    /**
++     * Get the value indicating the ticks spent
++     * breaking this door.
++     * This will return 0 when the breaking progress starts.
++     *
++     * @return The breaking progress value.
++     */
++    @Range(from = 0, to = Integer.MAX_VALUE)
++    public int getBreakTime() {
++        return this.breakTime;
++    }
++
++    /**
++     * Set the current progress in ticks that
++     * this entity has spent damaging the door.
++     *
++     * @param breakTime The ticks spent damaging the door.
++     */
++    public void setBreakTime(@Range(from = 0, to = Integer.MAX_VALUE) final int breakTime) {
++        Preconditions.checkArgument(breakTime >= 0, "The breaking progress must be at least -1!");
++        Preconditions.checkArgument(breakTime <= this.doorBreakTime, "The door breaking time cannot be greater than the amount of ticks required to break the door!");
++        this.breakTime = breakTime;
++    }
++
++    /**
++     * Get the total amount of ticks required to break
++     * the door that the entity is currently damaging.
++     *
++     * @return The amount of ticks necessary for the entity to break the door.
++     */
++    @Range(from = 0, to = Integer.MAX_VALUE)
```

Comment:
This return -1 for zombies so the current range is wrong but i think it would be better to call the getDoorBreakTime() that fallback to the default value of 240 ticks for the other constructor.